stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/MessageDispatcher.java

```diff
@@ -0,0 +1,214 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.messaging.rabbitmq;
+
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.AlreadyClosedException;
+import com.rabbitmq.client.Channel;
+import org.ballerinalang.jvm.BallerinaValues;
+import org.ballerinalang.jvm.JSONParser;
+import org.ballerinalang.jvm.JSONUtils;
+import org.ballerinalang.jvm.XMLFactory;
+import org.ballerinalang.jvm.types.AttachedFunction;
+import org.ballerinalang.jvm.types.BArrayType;
+import org.ballerinalang.jvm.types.BStructureType;
+import org.ballerinalang.jvm.types.BType;
+import org.ballerinalang.jvm.types.TypeTags;
+import org.ballerinalang.jvm.util.exceptions.BallerinaConnectorException;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.jvm.values.connector.Executor;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Handles and dispatched messages with data binding.
+ *
+ * @since 0.995
+ */
+public class MessageDispatcher {
+
+    private RabbitMQTransactionContext rabbitMQTransactionContext;
+    private Channel channel;
+    private boolean autoAck;
+    private ObjectValue service;
+
+    public MessageDispatcher(RabbitMQTransactionContext rabbitMQTransactionContext, ObjectValue service,
+                             Channel channel, boolean autoAck) {
+        this.rabbitMQTransactionContext = rabbitMQTransactionContext;
+        this.channel = channel;
+        this.autoAck = autoAck;
+        this.service = service;
+    }
+
+    /**
+     * Dispatch messages.
+     *
+     * @param message     Message content to be dispatched to the resource function.
+     * @param deliveryTag Delivery tag of the message.
+     * @param properties  Basic properties of the message.
+     */
+    public void handleDispatch(byte[] message, long deliveryTag, AMQP.BasicProperties properties) {
+        AttachedFunction[] attachedFunctions = service.getType().getAttachedFunctions();
+        AttachedFunction onMessageFunction;
+        if (RabbitMQConstants.FUNC_ON_MESSAGE.equals(attachedFunctions[0].getName())) {
+            onMessageFunction = attachedFunctions[0];
+        } else if (RabbitMQConstants.FUNC_ON_MESSAGE.equals(attachedFunctions[1].getName())) {
+            onMessageFunction = attachedFunctions[1];
+        } else {
+            return;
+        }
+        BType[] paramTypes = onMessageFunction.paramTypes;
+        int paramSize = paramTypes.length;
+        if (paramSize > 1) {
```

Comment:
Don't we need to check the param size is 2 here? If it is greater than 2 then we should return a panic. If this is already handled in compiler plugin, then this should be fine.

Reply:
Yes the compiler plugin is updated with this b12340b641b3eeee8abf3957ec408e646232d73c